### ISSUE TITLE
Allow command line configuration IConfiguration substitutions.

### ISSUE THREAD

**Issue by simkin2004** [mytrout/Pipelines#239]
On a Production Processor, I have two different connection strings that could be used, depending on the processor.

Rather than installing a second instance, I would prefer to substitute the IConfiguration variable named SecondaryDatabaseConnectionString for the DatabaseConnectionString value.

Since the substitution would take place at the Command line level, it would override even command line arguments.

Basically, the functionality returns the IConfiguration value referenced by the name SecondaryDatabaseConnectionString when a IConfiguration request for anything named DatabaseConnectionString is received.

perhaps something like 

`--substituteTarget DatabaseConnectionString --substituteValue SecondaryDatabaseConnectionString`
